src/utils/types/index.ts

```diff
@@ -0,0 +1,5 @@
+export type isAnyArray = (
```

Comment:
The type name "`isAnyArray`" can be misleading.
This is not a function that determines an array of type `any`, but a type of function that allows you to correctly determine the `ReadonlyArray` type.

Reply:
What name is appropriate?

Reply:
The TypeScript 3.5.2 `Array.isArray` function does not function as a [Type Guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types) for arrays of type `ReadonlyArray`.
A third party can not understand the need for the `isAnyArray` type without knowing this.

Reply:
> The TypeScript 3.5.2 `Array.isArray` function does not function as a [Type Guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types) for arrays of type `ReadonlyArray`.
> A third party can not understand the need for the `isAnyArray` type without knowing this.
> 

It is appropriate for the type name to make its purpose clear.
For example, what about `FixIsArray`?